azavea/pfb-network-connectivity
# Fixup local analysis

## Overview

This PR updates the `./scripts/run-local-analysis` script and the associated inner workings of the BNA analysis scripts to work entirely locally, with no connection to AWS or the development Django application. I was able to complete a Germantown network analysis locally without AWS credentials configured or the Django stack running.

### Notes

~~This PR does not yet incorporate any of the additional arguments provided by Spencer's replacement python script in https://github.com/azavea/pfb-network-connectivity/compare/develop...spencerrecneps:feature/better-local-analysis. It only takes the path of least resistance to a working local analysis within the current VM/Docker setup.~~ Edit: It now respects the `NB_POSTGRESQL_*` variables and supports targeting a remote DB.

## Testing Instructions

Follow the setup instructions in README.md, without configuring any AWS credentials. Then follow the instructions to run a job in README.LOCAL-ANALYSIS.md. The job should run successfully with no AWS credentials or PFB_JOB_ID configured.

If you already have a dev VM setup, and have AWS credentials configured, you can run a local analysis of the Germantown neighborhood in Philly with the following command:
```
AWS_PROFILE=doesnotexist NB_OUTPUT_DIR=/data/output/germantown-local \
    ./scripts/run-local-analysis \
    'https://andrew-pfb-storage-us-east-1.s3.amazonaws.com/neighborhood_boundaries/root/PA/germantown-philadelphia.zip' PA 42
```
In the above command, we must manually set AWS_PROFILE to an invalid profile that doesn't exist, to avoid using the correctly configured one.


Closes #574
